test/setup.tsx

```diff
@@ -23,18 +23,20 @@ function copyProps(src: any, target: any) {
   Object.defineProperties(target, propsAsPropertyDescriptorMap);
 }
 
-let temp: any = null;
-const localS = {
-  getItem() {
-    return temp;
-  },
-  setItem(key: any, value: any) {
-    temp = value;
-  },
-};
+function localStorage() {
+  let storage = {};
+  return {
+    getItem: function (key: string) {
+      return key in storage ? (storage as any)[key] : null;
+    },
+    setItem: function (key: string, value: any) {
+      (storage as any)[key] = value || '';
+    },
+  };
+}
```

Comment:
Den forrige funksjonen her hentet hva som helst som lå i `temp`. Siden hverken `set` eller `get` funksjonen brukte `key` så ville en verdi ble satt og den verdien returnert for alle som prøvde å bruke localstorage.